axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceServiceImpl.java

```diff
@@ -971,4 +985,87 @@ protected long getNotLetteredAdvancePaymentMoveLinesAmount(Partner partner) {
         .getResultList()
         .size();
   }
+
+  /**
+   * @param invoice
+   * @throws AxelorException
+   */
+  private void calculCA(Invoice invoice) throws AxelorException {
+
+    Partner partner = invoice.getPartner();
+    List<Partner> partnerParentList = new ArrayList<Partner>();
+    partnerParentList.add(partner);
+
+    /*
+     * Manage Date activity of civil year and fiscal year
+     * Search object in database corresponding to civil year and fiscal year compared to date of the day
+     * If Same date => one treatment (Civil year)
+     */
+    // Fiscal Year
+    Year yearFiscal;
+    try {
+      yearFiscal = yearServiceAccountImpl.getYear(YearBaseRepository.TYPE_FISCAL);
+    } catch (NoResultException e) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR, I18n.get(IExceptionMessage.YEAR_CIVIL));
+    }
+
+    // Civil year
+    Year yearCivil;
+    try {
+      yearCivil = yearServiceAccountImpl.getYear(YearBaseRepository.TYPE_CIVIL);
+    } catch (NoResultException e) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(IExceptionMessage.YEAR_FISCAL));
+    }
+
+    // Comparaison of start date and end date
+    boolean isSameDate =
+        (yearCivil.getFromDate().isEqual(yearFiscal.getFromDate()))
+            && (yearCivil.getToDate().isEqual(yearFiscal.getToDate()));
+
+    // Case of Supplier
+    if (invoice.getOperationTypeSelect() == 1 || invoice.getOperationTypeSelect() == 2) {
+      partnerTurnoverServiceImpl.calculCA(
+          partner, true, (isSameDate ? null : yearFiscal), yearCivil, partnerParentList);
+    }
+    // Case of Customer
+    else if (invoice.getOperationTypeSelect() == 3 || invoice.getOperationTypeSelect() == 4) {
+      partnerTurnoverServiceImpl.calculCA(
```

Comment:
Please use constant defined in InvoiceRepository instead of hard coded values for operation type select

Reply:
Done